src/Donations/Controllers/DonationRequestController.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Give\Donations\Controllers;
+
+use Give\Donations\Models\Donation;
+use Give\Donations\ValueObjects\DonationRoute;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+
+/**
+ * @unreleased
+ */
+class DonationRequestController
+{
+    /**
+     * @unreleased
+     *
+     * @return WP_Error | WP_REST_Response
+     */
+    public function getDonation(WP_REST_Request $request)
+    {
+        $donation = Donation::find($request->get_param('id'));
+
+        if ( ! $donation) {
+            return new WP_Error('donation_not_found', __('Donation not found', 'give'), ['status' => 404]);
+        }
+
+        return new WP_REST_Response($this->escDonation($donation));
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonations(WP_REST_Request $request): WP_REST_Response
+    {
+        $page = $request->get_param('page');
+        $perPage = $request->get_param('per_page');
+        $sortColumn = $this->getSortColumn($request->get_param('sort'));
+        $sortDirection = $request->get_param('direction');
+
+        $query = Donation::query();
+
+        if ($campaignId = $request->get_param('campaignId')) {
+            // Filter by CampaignId
+            $query->where('give_donationmeta_attach_meta_campaignId.meta_value', $campaignId);
+        }
+
+        if ($request->get_param('hideAnonymousDonations')) {
+            // Exclude anonymous donations from results
+            $query->where('give_donationmeta_attach_meta_anonymous.meta_value', 0);
+        }
```

Comment:
I recommend keeping parameters in the positive as it avoids double negatives (I want to not hide...). Let's rename this to `includeAnonymousDonations` and make it true by default.

Reply:
Resolved in ded974c38462d50bb6a55d9a2d2e9aa5a7a6aa8e